mlflow/pyspark/ml/__init__.py

```diff
@@ -257,29 +500,101 @@ def _log_pretraining_metadata(estimator, params):
+                _get_instance_param_map_recursively(
+                    param_search_estimator.getEstimator(), 1, autologging_metadata.uid2name_map
+                )
+
+        if artifact_dict:
             try_mlflow_log(
-                mlflow.log_dict, pipeline_hierarchy, artifact_file="pipeline_hierarchy.json"
+                mlflow.log_dict, artifact_dict, artifact_file="estimator_info.json"
             )
 
         # Chunk model parameters to avoid hitting the log_batch API limit
-        for chunk in _chunk_dict(param_map, chunk_size=MAX_PARAMS_TAGS_PER_BATCH,):
+        for chunk in _chunk_dict(
+            param_map, chunk_size=MAX_PARAMS_TAGS_PER_BATCH,
+        ):
             truncated = _truncate_dict(chunk, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
             try_mlflow_log(mlflow.log_params, truncated)
 
         try_mlflow_log(mlflow.set_tags, _get_estimator_info_tags(estimator))
 
     def _log_posttraining_metadata(estimator, spark_model, params):
-        # TODO: Log nested runs for spark ml tuning estimators
-        #   (CrossValidator/TrainValidationSplit)
+
+        if _is_parameter_search_estimator(estimator):
+            try:
+                # Fetch environment-specific tags (e.g., user and source) to ensure that lineage
+                # information is consistent with the parent run
+                child_tags = context_registry.resolve_tags()
+                child_tags.update({MLFLOW_AUTOLOGGING: AUTOLOGGING_INTEGRATION_NAME})
+                _create_child_runs_for_parameter_search(
+                    parent_estimator=estimator,
+                    parent_model=spark_model,
+                    parent_run=mlflow.active_run(),
+                    child_tags=child_tags
+                )
+            except Exception as e:
+                import traceback
+                msg = (
+                    "Encountered exception during creation of child runs for parameter search."
+                    " Child runs may be missing. Exception: {}".format(traceback.format_exc())
+                )
+                _logger.warning(msg)
+
+            estimator_param_maps = _get_tuning_param_maps(
+                estimator, estimator._autologging_metadata.uid2name_map
+            )
+
+            metric_key, metrics = _get_param_search_metrics(estimator, spark_model)
+            _log_parameter_search_results_as_artifact(
+                estimator_param_maps,
+                metrics,
+                metric_key,
+                mlflow.active_run().info.run_id
+            )
+
+            if estimator.getEvaluator().isLargerBetter():
+                best_index = np.argmax(metrics)
+            else:
+                best_index = np.argmin(metrics)
+
+            best_param_map = estimator_param_maps[best_index]
+            try_mlflow_log(
+                mlflow.log_dict,
+                best_param_map,
+                artifact_file="best_parameters.json"
+            )
```

Comment:
Can we log these as params on the run as well?

For example, given the following contents of `best_parameters.json`:

```
{
  "LogisticRegression.maxIter": "10",
  "LogisticRegression.regParam": "0.1"
}
```

can we log the parameters: `best_maxIter = 10` and `best_regParam = 0.1` to MLflow tracking?